edgelet/edgelet-hsm/tests/crypto_master_key_success.rs

```diff
@@ -3,12 +3,22 @@
 #![deny(unused_extern_crates, warnings)]
 #![deny(clippy::all, clippy::pedantic)]
 
+use std::env;
+use tempfile::TempDir;
+
 use edgelet_core::crypto::MasterEncryptionKey;
 use edgelet_hsm::Crypto;
 
+const HOMEDIR_KEY: &str = "IOTEDGE_HOMEDIR";
+
 /// Encryption master key tests
 #[test]
 fn crypto_master_key_success() {
+    // arrange
+    let home_dir = TempDir::new().unwrap();
+    env::set_var(HOMEDIR_KEY, &home_dir.path());
```

Comment:
`#[test]` tests run in parallel by default, so if anyone adds another test later that also wants to set an env var there'll be trouble.

Can you change the API to take an `Option<PathBuf>` that defaults to `env::var(HOMEDIR_KEY)` instead? Then the test can pass in `Some(home_dir.path())` and the actual code can pass in `None`

Same for the other tests that set env vars.

Reply:
Fixed thanks for the suggestion.